src/Nethermind/Nethermind.Merge.Plugin/Handlers/NewPayloadHandler.cs

```diff
@@ -164,7 +169,11 @@ public async Task<ResultWrapper<PayloadStatusV1>> HandleAsync(ExecutionPayload r
             return NewPayloadV1Result.Syncing;
         }
 
-        NewPayloadForParentReceived?.Invoke(this, parentHeader);
+        if (_simulateBlockProduction)
+        {
+            _payloadPreparationService.CancelBlockProduction(parentHeader.GenerateSimulatedPayload()
+                .GetPayloadId(parentHeader));
+        }
```

Comment:
I would keep it as it was and wire it in the plugin? Or with another component that takes NewPayloadHandler and PayloadPreparationService?

Reply:
I'd prefer another component or something that wraps `NewPayloadHandler`. But the whole thing is sketchy.